### ISSUE TITLE
Logger facility

### ISSUE THREAD

**Issue by thibault-martinez** [iotaledger/entangled#77]
Currently, two logging facilities are being used in different branches.

- https://github.com/embear/logger that turned out not being thread-safe so locks were added on top of it
- A thread-safe logger facility that was made from scratch

This issue aims at merging the best out of the two current logger facilities

**Comment by tsvisabo** [comment]
Maybe @th0br0  can be the judge of it ?

**Comment by tsvisabo** [comment]
@thibault-martinez 
I think we should go with `embear`'s for now since it has been there for a while and is fully tested +
it has contexts support (`logger_id_t`)
can register multiple files (including stdout/stderr) to different error levels, and is highly configurable including prefixing each log with date/line/function etc.. and colors..

**Comment by thibault-martinez** [comment]
@tsvisabo Totally agree

**Comment by thibault-martinez** [comment]
@tsvisabo I would add macros on top of that though.

Something like

```
#define log_trace(...) logger_log(LOG_TRACE, __VA_ARGS__)
#define log_debug(...) logger_log(LOG_DEBUG, __VA_ARGS__)
#define log_info(...) logger_log(LOG_INFO, __VA_ARGS__)
#define log_warn(...) logger_log(LOG_WARN, __VA_ARGS__)
#define log_error(...) logger_log(LOG_ERROR, __VA_ARGS__)
#define log_fatal(...) logger_log(LOG_FATAL, __VA_ARGS__)
```

I find the name and need to specify the level quite verbose as we are going to abuse it.

`logger_helper_print(CONNECTION_LOGGER_ID, LOGGER_ERR, "Failed in creating index, statement: %s", statement);`

What do you think ?

**Comment by tsvisabo** [comment]
@thibault-martinez 
sounds good

**Comment by thibault-martinez** [comment]
Closed by #127
